internal/service/cluster/cluster.go

```diff
@@ -246,11 +253,17 @@ func (s *Service) Run(ctx context.Context, host service.Host) error {
 
 	peers, err := s.getPeers()
 	if err != nil {
-		return fmt.Errorf("failed to get peers to join: %w", err)
+		// We can continue here as we have a rejoin goroutine try again later.
+		level.Warn(s.log).Log("msg", "failed to get peers to join at startup", "err", err)
+		peers = nil
```

Comment:
> In general I think it's a good idea to fail fast, especially that many users don't have robust metamonitoring etc. It seems you're on-board with this so I will change it to fail fast in both cases.

As long as "both cases" doesn't mean failing fast whenever we do a rediscovery in the background, I'm on board 👍

Reply:
Cool, I've changed it to fail fast on satartup. It will not fail on rediscovery in the rejoin routine - [it will log a warning](https://github.com/grafana/alloy/blob/fdddc44bfb2b740dc3fa0911f8e930ce663f52f1/internal/service/cluster/cluster.go#L289) and continue trying.